src/components/utils/test/posix_thread_test.cc

```diff
@@ -41,287 +43,505 @@ namespace utils_test {
   // Arrange
   threads::Thread* thread = NULL;
   TestThreadDelegate* threadDelegate = new TestThreadDelegate();
-  AutoLock test_lock(test_mutex_);
-  // Create thread
-  ASSERT_NO_THROW(thread = CreateThread(threadName, threadDelegate));
-  // Start thread with default options (Stack size = 0 & thread is detached)
-  thread->start(threads::ThreadOptions(0, false));
-  // Check thread is detached
-  EXPECT_FALSE(thread->is_joinable());
-  // Check thread stack size is 0
-  EXPECT_EQ(Thread::kMinStackSize, thread->stack_size());
-  cond_var_.WaitFor(test_lock, 10000);
+
+  {
+    AutoLock test_lock(*threadDelegate);
+    // Create thread
+    ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+    // Start thread with default options (Stack size = 0 & thread is detached)
+    thread->start(threads::ThreadOptions(0, false));
+    // Check thread is detached
+    EXPECT_FALSE(thread->is_joinable());
+    // Check thread stack size is 0
+    EXPECT_EQ(Thread::kMinStackSize, thread->stack_size());
+    threadDelegate->WaitFor(test_lock, 10000);
+  }
+
   EXPECT_TRUE(threadDelegate->check_value());
-  DeleteThread(thread);
+
   delete threadDelegate;
   EXPECT_EQ(NULL, thread->delegate());
+  DeleteThread(thread);
 }
 
-TEST(
-    PosixThreadTest,
-    DISABLED_CheckCreatedThreadNameChangeToEmpty_ExpectThreadNameChangedToEmpty) {
+TEST_F(PosixThreadTest,
+       CheckCreatedThreadNameChangeToEmpty_ExpectThreadNameChangedToEmpty) {
   // Arrange
   threads::Thread* thread = NULL;
   TestThreadDelegate* threadDelegate = new TestThreadDelegate();
-  AutoLock test_lock(test_mutex_);
-  // Create thread
-  ASSERT_NO_THROW(thread = CreateThread(threadName, threadDelegate));
-  thread->start(threads::ThreadOptions(threads::Thread::kMinStackSize));
-  // Rename started thread. Name will be cut to 15 symbols + '\0'
-  // This is the limit in current POSIX thread implementation
-  thread->SetNameForId(thread->thread_handle(), std::string(""));
-  // Name must be large enough to keep 16 symbols. Read previous comment
-  char name[MAX_SIZE];
-  int result = pthread_getname_np(thread->thread_handle(), name, sizeof(name));
-  if (!result) {
-    EXPECT_EQ(std::string(""), std::string(name));
+
+  {
+    AutoLock test_lock(*threadDelegate);
+    // Create thread
+    ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+    thread->start(threads::ThreadOptions(threads::Thread::kMinStackSize));
+    // Rename started thread. Name will be cut to 15 symbols + '\0'
+    // This is the limit in current POSIX thread implementation
+    thread->SetNameForId(thread->thread_handle(), std::string(""));
+    // Name must be large enough to keep 16 symbols. Read previous comment
+    char name[kMaxSize];
+    int result =
+        pthread_getname_np(thread->thread_handle(), name, sizeof(name));
+    if (!result) {
+      EXPECT_EQ(std::string(""), std::string(name));
```

Comment:
```suggestion
      EXPECT_TRUE(std::string(name).empty());
```

Reply:
@YarikMamykin 
Thanks, fixed in:https://github.com/LuxoftSDL/sdl_core/pull/50/commits/c3ce3eb57db8a82ab440a6442ec6cc9fcb98129e